control-plane/api-gateway/binding/result.go

```diff
@@ -382,59 +383,75 @@ func (l listenerValidationResult) conflictedCondition(generation int64) metav1.C
 }
 
 // acceptedCondition constructs the condition for the ResolvedRefs status type.
-func (l listenerValidationResult) resolvedRefsCondition(generation int64) metav1.Condition {
+func (l listenerValidationResult) resolvedRefsConditions(generation int64) []metav1.Condition {
 	now := timeFunc()
 
+	conditions := make([]metav1.Condition, 0)
+
 	if l.routeKindErr != nil {
-		return metav1.Condition{
+		return []metav1.Condition{{
 			Type:               "ResolvedRefs",
 			Status:             metav1.ConditionFalse,
 			Reason:             "InvalidRouteKinds",
 			ObservedGeneration: generation,
 			Message:            l.routeKindErr.Error(),
 			LastTransitionTime: now,
-		}
+		}}
 	}
 
-	switch l.refErr {
-	case errListenerInvalidCertificateRef_NotFound, errListenerInvalidCertificateRef_NotSupported, errListenerInvalidCertificateRef_InvalidData, errListenerInvalidCertificateRef_NonFIPSRSAKeyLen, errListenerInvalidCertificateRef_FIPSRSAKeyLen:
-		return metav1.Condition{
-			Type:               "ResolvedRefs",
-			Status:             metav1.ConditionFalse,
-			Reason:             "InvalidCertificateRef",
-			ObservedGeneration: generation,
-			Message:            l.refErr.Error(),
-			LastTransitionTime: now,
-		}
-	case errRefNotPermitted:
-		return metav1.Condition{
-			Type:               "ResolvedRefs",
-			Status:             metav1.ConditionFalse,
-			Reason:             "RefNotPermitted",
-			ObservedGeneration: generation,
-			Message:            l.refErr.Error(),
-			LastTransitionTime: now,
+	for _, refErr := range l.refErrs {
+		switch refErr {
+		case errListenerInvalidCertificateRef_NotFound, errListenerInvalidCertificateRef_NotSupported, errListenerInvalidCertificateRef_InvalidData, errListenerInvalidCertificateRef_NonFIPSRSAKeyLen, errListenerInvalidCertificateRef_FIPSRSAKeyLen:
```

Comment:
For ease of reading: 
```suggestion
		case errListenerInvalidCertificateRef_NotFound,
		     errListenerInvalidCertificateRef_NotSupported,
		     errListenerInvalidCertificateRef_InvalidData,
		     errListenerInvalidCertificateRef_NonFIPSRSAKeyLen,
		     errListenerInvalidCertificateRef_FIPSRSAKeyLen:
```